### PR TITLE
Fix typos discovered by codespell

### PR DESCRIPTION
**Pull request by cclauss**
This PR was created with the command [`codespell --ignore-words-list="ba,datas,fo,uptodate" -w`](https://github.com/codespell-project/codespell)

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: tests/unit/test_updater.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def test_check_update_respect_config(mock_check, result, updater, mocker):
         ("0.0.2", "0.0.3", True),
         ("0.0.2", "0.0.1", False),
     ],
-    ids=["uptodate", "behind", "ahead"],
+    ids=["up-to-date", "behind", "ahead"],
```
Comment (skshetry) [inline tests/unit/test_updater.py:86]: I'd prefer the previous one, as it is easier to type in the terminal, in case we need to run that particular test.

**Comment by skshetry** [comment]
`codespell` seems to offer a pre-commit hook as well. @cclauss, what do you think? How is it in terms of false-positives?

@iterative/dvc WDYT?

File: tests/unit/test_updater.py (hunk 1 above)
Comment (cclauss) [reply tests/unit/test_updater.py]: Reverted.

**Comment by cclauss** [comment]
The pre-commit works well and the syntax `codespell—ignore-words-list="uptodate”` can deal with any false reports.

**Review by pared — COMMENTED** [review]
LGTM, seems like we could add it to pre-commit.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -7,6 +7,12 @@ repos:
         language_version: python3
     repo: https://github.com/ambv/black
     rev: 21.6b0
+  - hooks:
+      - id: codespell
+        args: 
+          - ignore-words-list ba,datas,fo,uptodate
```
Comment (skshetry) [inline .pre-commit-config.yaml:13]: I think you need to split this into two:
```suggestion
          - ignore-words-list
          - ba,datas,fo,uptodate
```

But let's see if this also works first though.

**Review by skshetry — APPROVED** [review]
Thanks @cclauss for the PR. 🙂

**Review by pared — APPROVED** [review]

**Pull request merged**